contracts/upgradable/Proxy.sol

```diff
@@ -6,7 +6,23 @@ import { IProxy } from '../interfaces/IProxy.sol';
 import { IUpgradable } from '../interfaces/IUpgradable.sol';
 import { BaseProxy } from './BaseProxy.sol';
 
+/**
+ * @title Proxy Contract
+ * @author Kiryl Yermakou
+ * @notice A proxy contract that delegates calls to a designated implementation contract. Inherits from BaseProxy.
+ * @dev The constructor takes in the address of the implementation contract, the owner address, and any optional setup
+ * parameters for the implementation contract.
+ */
 contract Proxy is BaseProxy {
+    /**
+     * @notice Constructs the proxy contract with a the implementation address, owner address, and optional setup parameters.
+     * @param implementationAddress The address of the implementation contract
+     * @param owner The owner address
+     * @param setupParams Optional parameters to setup the implementation contract
+     * @dev The constructor verifies that the owner address is not the zero address and that the contract ID of the implementation is valid.
+     * It then stores the implementation address and owner address in their designated storage slots and calls the setup function on the
+     * implementation (if setup params exist).
+     */
     constructor(
         address implementationAddress,
         address owner,
```

Comment:
avoid low level call warning can also be ignored in solhint

Reply:
added to solhint